File: manual/code/test_suites/python/python/test_Polars_sum.py
import unittest
from function_Polars_sum import PolarsExample

class TestPolarsExample(unittest.TestCase):

    def test_compute_sum_multiple_positive_numbers(self):
        calc = PolarsExample()
        result = calc.compute_sum(1, 2, 3)
        self.assertEqual(result, 6.0)

    def test_compute_sum_multiple_negative_numbers(self):
        calc = PolarsExample()
        result = calc.compute_sum(-1, -2, -3)
        self.assertEqual(result, -6.0)

    def test_compute_sum_mixed_numbers(self):
        calc = PolarsExample()
        result = calc.compute_sum(1, -1, 2)
        self.assertEqual(result, 2.0)

    def test_compute_sum_single_number(self):
        calc = PolarsExample()
        result = calc.compute_sum(10)
        self.assertEqual(result, 10.0)

    def test_compute_sum_no_input(self):
        calc = PolarsExample()
        result = calc.compute_sum(10, None, None)
        self.assertEqual(result, 10.0)  # Only 10 should be considered

if __name__ == '__main__':
    unittest.main()

File: code/test_suites/cpp/FunctionBuildTest/tests/test_Polars_sum.cpp
#include <gtest/gtest.h>
#include "../src/function_Polars_sum.cpp"

class TestPolarsExample : public ::testing::Test {
protected:
    PolarsExample calc;
};

TEST_F(TestPolarsExample, TestComputeSumMultiplePositiveNumbers) {
    float result = calc.compute_sum(1, 2, 3);
    EXPECT_FLOAT_EQ(result, 6.0f);
}

TEST_F(TestPolarsExample, TestComputeSumMultipleNegativeNumbers) {
    float result = calc.compute_sum(-1, -2, -3);
    EXPECT_FLOAT_EQ(result, -6.0f);
}

TEST_F(TestPolarsExample, TestComputeSumMixedNumbers) {
    float result = calc.compute_sum(1, -1, 2);
    EXPECT_FLOAT_EQ(result, 2.0f);
}

TEST_F(TestPolarsExample, TestComputeSumSingleNumber) {
    float result = calc.compute_sum(10);
    EXPECT_FLOAT_EQ(result, 10.0f);
}

TEST_F(TestPolarsExample, TestComputeSumNoInput) {
    float result = calc.compute_sum(10, boost::none, boost::none);
    EXPECT_FLOAT_EQ(result, 10.0f);
}

int main(int argc, char **argv) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}